spec/unit/services_inspector_spec.rb

```diff
@@ -187,5 +187,40 @@
         Service.new(name: "dnsmasq", state: "off")
       ])
     end
+  
+    it "prints a warning when chkconfig has xinetd output" do
+      chkconfig_redhat_xinetd_output =
+        <<-EOF
+crond 0:off 1:off 2:on 3:on 4:on 5:on 6:off
+dnsmasq 0:off 1:off 2:off 3:off 4:off 5:off 6:off
+xinetd based services:
+    chargen-dgram:  off
+    chargen-stream: off
+EOF
+
+      allow(system).to receive(:has_command?).
+        with("systemctl").and_return(false)
+      allow(system).to receive(:has_command?).
+        with("chkconfig", "--version").and_return(true)
+      
+      expect(system).to receive(:check_requirement).
+        with("chkconfig", "--version")
+      expect(system).to receive(:check_requirement).
+        with("runlevel")
+      expect(system).to receive(:run_command).
+        with("runlevel", stdout: :capture).
+        and_return("N 5")
+      expect(system).to receive(:run_command).
+        with("chkconfig", "--list", stdout: :capture).
+      and_return(chkconfig_redhat_xinetd_output)
+
+      expect(Machinery::Ui).to receive(:warn)
+      services = inspector.send(:parse_redhat_chkconfig)
```

Comment:
Lines 218 to 223 look redundant as they are covered be the test above and are not needed for testing if a warning is shown. I don't insist on removing these lines, though.